mlflow/tensorflow/callback.py

```diff
@@ -101,3 +103,98 @@ def on_test_end(self, logs=None):
             return
         metrics = {"validation_" + k: v for k, v in logs.items()}
         log_metrics(metrics, synchronous=False)
+
+
+class MlflowModelCheckpointCallback(Callback, MlflowModelCheckpointCallbackBase):
+    """Callback for automatic Keras model checkpointing to MLflow.
+    """
+
+    def __init__(
+        self,
+        client,
+        run_id,
+        monitor="val_loss",
+        mode="min",
+        save_best_only=True,
+        save_weights_only=False,
+        save_freq="epoch",
+    ):
+        """
+        Args:
+            client: An instance of `MlflowClient`.
+            run_id: The id of the MLflow run which you want to log checkpoints to.
+            monitor: In automatic model checkpointing, the metric name to monitor if
+                you set `model_checkpoint_save_best_only` to True.
+            save_best_only: If True, automatic model checkpointing only saves when
+                the model is considered the "best" model according to the quantity
+                monitored and previous checkpoint model is overwritten.
+            mode: one of {"min", "max"}. In automatic model checkpointing,
+                if save_best_only=True, the decision to overwrite the current save file is made
+                based on either the maximization or the minimization of the monitored quantity.
+            save_weights_only: In automatic model checkpointing, if True, then
+                only the model’s weights will be saved. Otherwise, the optimizer states,
+                lr-scheduler states, etc are added in the checkpoint too.
+            save_freq: `"epoch"` or integer. When using `"epoch"`, the callback
+                saves the model after each epoch. When using integer, the callback
+                saves the model at end of this many batches. Note that if the saving isn't
+                aligned to epochs, the monitored metric may potentially be less reliable (it
+                could reflect as little as 1 batch, since the metrics get reset
+                every epoch). Defaults to `"epoch"`.
+        """
+        Callback.__init__(self)
+        MlflowModelCheckpointCallbackBase.__init__(
+            self,
+            client=client,
+            run_id=run_id,
+            checkpoint_file_suffix="h5",
+            monitor=monitor,
+            mode=mode,
+            save_best_only=save_best_only,
+            save_weights_only=save_weights_only,
+            save_freq=save_freq,
+        )
+        self.trainer = None
```

Comment:
do we need this? Keras callbacks will be attached to Keras model, so it automatically gets access to the model instance.